draft-wing-cidfi.md

```diff
@@ -1183,42 +1185,9 @@ thus maintaining the end user's privacy.  Communications are relayed through the
-on-path CIDFI network element (which needs to see the Discovery
-Packet) the attacker can then 'steal' the victim's CIDFI control from
-the victim's 5-tuple and the victim's CIDFI signaling thereafter for
-that 5-tuple will influence the attacker's 5-tuple.  The attacker
-can't see that CIDFI signaling (because it still goes to the victim
-which relays the CIDFI signaling to the CIDFI network element) but
-with this attack the victim's CIDFI treatment is effectively disabled
-and is available to the attacker.  The attacker can send
-NEW_CONNECTION_ID frames to the server with the victim's (observed)
-Destination CID, effectively stealing the victim's CIDFI signaling for
-themselves.  To mitigate this replay attack shared networks need
-per-endpoint encryption (e.g., Wi-Fi WPA3, DOCSIS BPI+) or traffic
-separation (e.g., Ethernet switching rather than bridging).
-
+Spoofing Attack: For an attacker to succeed with the nonce challenge against a victim's UDP 4-tuple, the attacker has to send a STUN CIDFI-NONCE packet using the victim's source IP address and a valid HMAC. A valid HMAC can only be obtained by the attacker making its own connection to the CIDFI server and spoofing the source IP address and UDP port of the victim. If the client does not support CIDFI, the attacker can influence the packet treatment of the victim's 5-tuple. If the client implements CIDFI, the CIDFI network element can identify an IP address spoofing attack . The CIDFI network element receives two HTTPS connections describing the same DCID (one connection from the attacker, another from the victim). The CIDFI network element will issue unique Nonces and HMACs to both the attacker and victim, and both the attacker and victim will send the STUN indication on that same UDP 4-tuple. It should generate an alarm on the CIDFI network element. In this scenario, it is recommended that both the attacker and the victim be denied CIDFI access. Such spoofing of a victim's IP address is prevented by the network using network ingress filtering ({{!RFC2827}}, {{!RFC7513}}, {{!RFC6105}}, and/or {{!RFC6620}}).
 
+On-Path Attack: An on-path attacker can observe the victim's Discovery Packet, block it, and then forward the packet within the attacker's 5-tuple. Subsequently, the on-path attacker can 'steal' the victim's CIDFI control from the victim's 5-tuple, causing the victim's CIDFI signaling for that 5-tuple to influence the attacker's 5-tuple. Although the on-path attacker can't directly observe the encrypted CIDFI signaling, this attack effectively disables the victim's CIDFI treatment, making it accessible to the attacker. The attacker can send NEW_CONNECTION_ID frames to the server with the victim's (observed) Destination CID, effectively claiming the victim's CIDFI signaling for themselves. To mitigate this replay attack, shared networks require per-endpoint encryption (e.g., Wi-Fi WPA3, DOCSIS BPI+) or traffic separation (e.g., Ethernet switching rather than bridging).
```

Comment:
> The attacker can send NEW_CONNECTION_ID frames to the server with the..

But this will be rejected by the server.

Reply:
A QUIC client is free to use use any DCID; multiple QUIC clients within the the same endpoint itself can use the same DCID.

Reply:
My comment was about an attacker who sends a NEW_CONNECTION_ID. Adding or retiring CIDs is encrypted and thus don't see on which case an attacker can inject those and pass through the validation at the server side.

Reply:
The on-path attacker can create a new QUIC connection using the same DCID.

Reply:
But if creates a new connection, then the initial DCID will be set to zero. 

Also, how this interferes with the connection that was established by the client.

Reply:
The on-path attacker creates a new QUIC connection with the same DCID as the victim to server, it drops the STUN indication message from client and sends STUN indication message to prove ownership of the 5-tuple. It means, the victim will not be able to prioritize packets. The connection from the victim will continue to exit.

Reply:
The relay attack is not applicable with a on-path attacker who is capable of dropping the traffic from victim. I will remove it.

Reply:
Thanks for removing that part. 

Still need a figure with the various messages to digest the attack.

Reply:
[cidfi-attack.pptx](https://github.com/danwing/cidfi/files/13437765/cidfi-attack.pptx)
Attached figure explaining the on-path attack.

Reply:
Thanks. The file points to the generic cidfi prez.

Reply:
[cidfi-attack.pptx](https://github.com/danwing/cidfi/files/13438288/cidfi-attack.pptx)

Reply:
Please see the updated presentation.